src/validateWebSite.ts

```diff
@@ -24,78 +23,57 @@ type WebError = {
+        const statusCodes: { code: number, elapsed: number, reported?: boolean }[] = [];
 
         // tslint:disable-next-line:no-constant-condition
         while (true) {
-            let isSuccess: boolean;
             try {
                 const response = <IncomingMessage>(await requestPromise(options));
-                // request throws an error for 400-500 responses
-                // a status code 200-300 indicates success
                 currentStatusCode = response.statusCode;
-                isSuccess = response.statusCode >= 200 && response.statusCode < 400;
-                currentStatusMessage = response.statusMessage;
-            } catch (e) {
-                const error = <WebError>e;
-                const response = error.response || {};
-                isSuccess = false;
+            } catch (error) {
+                const response = (<WebError>error).response || {};
                 currentStatusCode = isNumber(response.statusCode) ? response.statusCode : 0;
-                currentStatusMessage = response.statusMessage || parseError(error).message;
             }
 
-            properties.statusCodes = properties.statusCodes ? `${properties.statusCodes},` : '';
-            properties.statusCodes += currentStatusCode;
-            properties.lastStatusCode = currentStatusCode.toString();
-            properties.lastStatusMessage = currentStatusMessage;
+            const elapsedSeconds = Math.round((Date.now() - start) / 1000);
+            statusCodes.push({ code: currentStatusCode, elapsed: elapsedSeconds });
 
-            if (isSuccess) {
-                log(siteWrapper, outputChannel, `${uri} returned successful status code ${currentStatusCode}`);
-                return;
+            if (Date.now() > start + maximumValidationMs) {
+                break;
             }
 
-            if (Date.now() > timeoutTime) {
-                properties.timedOut = 'true';
-                log(siteWrapper, outputChannel, `Timed out waiting for successful response from ${uri}. Last status code returned: ${currentStatusCode}`);
-                throw new Error(currentStatusMessage);
-            }
-
-            // tslint:disable-next-line:no-string-based-set-timeout // false positive
-            await new Promise<void>((resolve, _reject): void => { setTimeout(resolve, pollingIntervalMs); });
+            await delay(pollingIntervalMs);
             pollingIntervalMs += pollingIncrementMs;
         }
+
+        properties.statusCodes = JSON.stringify(statusCodes);
     });
 }
 
-function log(siteWrapper: SiteWrapper, outputChannel: OutputChannel, message: string): void {
-    outputChannel.appendLine(`${(new Date()).toLocaleTimeString()} ${siteWrapper.appName}: ${message}`);
+async function delay(delayMs: number): Promise<void> {
+    // tslint:disable-next-line:no-string-based-set-timeout // tslint false positive
+    await new Promise<void>((resolve, _reject): void => { setTimeout(resolve, delayMs); });
```

Comment:
nit: You don't have to disable the rule if you specify the type of resolve. Also you don't have to include 'reject' at all:
```
await new Promise((resolve: () => void): void => { setTimeout(resolve, delayMs); })
```